src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1035,6 +1069,14 @@ namespace xclcpuemhal2 {
   
   CpuemShim::~CpuemShim() 
   {
+    if (mIsKdsSwEmu && mSWSch && mCore)
+    {
+      mSWSch->fini_scheduler_thread();
+      delete mCore;
```

Comment:
deletes look suspicious.  should this heap variables be managed in a smart pointer?

Reply:
this line gets called during end of program execution.